doc/adr/0010-project-versioning-and-release-flow.md

```diff
@@ -0,0 +1,56 @@
+# 10. Project versioning and release flow
+
+Date: 2022-05-26
+
+## Status
+
+Proposed
+
+## Context
+
+We need a versioning scheme for the project that will allow simultaneous development of the library modules (`kernel` and `sdk`) and the projects in the `examples/` directory that can depend 
+on the library modules. SNAPSHOT versioning will allow this.
+
+## Decision
+
+1. `master` branch contains changes that have been released.
+2. `develop` branch contains changes that haven't been released yet.
+3. Project versioning follows [semver](https://semver.org).
+4. `master` branch must contain version without precedence (`MAJOR.MINOR.PATCH`) - version `a.b.c-alpha` or `a.b.c-SNAPSHOT` is not allowed.
+5. `develop` branch should reflect the next development version of the latest
+   release, so `MAJOR.MINOR+1.0-SNAPSHOT`.
+
+Example: when the latest released version is `v0.5.0`, the project version on `master` branch is `v0.5.0` while 
+on `develop` branch, the project version is `v0.6.0-SNAPSHOT`.
+
+6. For merging release or hotfix changes back into master or develop, `git merge` should be used, with fast-forward when possible.
+
+### Release flow
+
+1. Create branch from the `develop` or feature branch: `release/vA.B.C`.
+2. Create commit with updated version for the release. (example: for development
+   version `v1.5.0-SNAPSHOT` create a commit message `v1.5.0` with updated [version.sbt](../../version.sbt))
+3. Merge release branch to `master` using `git merge`.
+4. Create tag reflecting released version: `git tag vA.B.C`.
+5. Push changes to origin `git push origin master --tags`.
+6. Merge `master` branch to `develop` using `git merge`.
+7. Create commit with an updated development version on `develop` branch.
+   (example: released `v1.5.0`, create commit `v1.6.0-SNAPSHOT`)
+8. Push changes to origin `git push origin develop`.
```

Comment:
It would be nice to automate that by some tool. Previously we have been doing that manually and so many points can lead to mistakes.

Reply:
That's true but I'd keep that original way written down, as it's exactly how we release and hotfix things currently. If we will have time for automation, I'd be more than happy to do this